jax/_src/lax/lax.py

```diff
@@ -2452,10 +2446,29 @@ def _bessel_i1e_jvp(g, y, x):
 imag_p = unop(_complex_basetype, _complex, 'imag')
 ad.deflinear2(imag_p, lambda t, _: [complex(np.zeros((), _dtype(t)), neg(t))])
 
+
+def _complex_transpose_rule(t, x, y):
```

Comment:
It kind of feels like some of this logic should be able to live in primitive dispatch (i.e., that primitives applied to Zeros should return Zeros)?

Reply:
Well, zeros currently aren't jax types, so we can't handle this in primitive application as primitives don't know about zeros. Instead, zeros are internal to the AD machinery, and have to be handled in the AD rules. That has some nice advantages but maybe some disadvantages too?

Are you +1ing the general idea of thinking about adding symbolic zeros as jax types (rather than having them internal to the AD machinery), or are you suggesting a different change for this PR? (Or maybe I've misunderstood!)

Reply:
Just the former!